AprilSylph/XKit-Rewritten
# Mutual Checker, NotificationBlock, Quote Replies: Process new reply format

Well, #1658 is fairly problematic.

### Description


This contains (somewhat rudimentary) fixes for the linked issue, in which the relevant features are broken by certain kinds of activity item that use a new internal data format, containing only enough information to render the notification rather than all of the internal data we use to add functionality.

- Mutual Checker can still easily determine if the replying user is a mutual, since Tumblr renders that.
- NotificationBlock wants to know the root post of the target post, which just so happens to work in this case because the activity items in question are on original posts, but will be a problem in the future if all activity items switch to this API.
- As per below discussion, with some slightly hairy URL and NPF parsing, Quote Replies can look up and fetch the correct reply and get the summary text from the notification data.

<details>
<summary>outdated Quote Replies info</summary>

- Quote Replies... is a somewhat complicated scenario because we **might** technically have enough information to look up the relevant post and post activity using an API fetch, but only via very convoluted methods (the activity item links to the post; we could parse the url to get the relevant information to API fetch the post but we have no timestamp with which to easily find the reply). For now, just to make it work on all posts rather than keep getting confused users, I've processed the activity item itself into a rudimentary post body:

<img width="561" src="https://github.com/user-attachments/assets/4ed41e98-0164-4925-9a57-28474ace609e">

This truncates the reply text and generally looks rather stupid, but it's better than nothing. Note also that it does not currently exclude community posts (I didn't see a way to do so). I did test the code on a reply on a community post.

</details>

### Testing steps


To easily distinguish affected activity items, run `document.documentElement.append(Object.assign(document.createElement('style'), { textContent: '[aria-label="Notification"]:not([role="listitem"]) { outline: 1px solid red; outline-offset: -3px }' }))`.

Mutual Checker:
- Enable the "only show notifications from mutuals" option.
- Confirm that from-mutuals notifications of both the old and new types are visible.
- Confirm that other notifications of both the old and new types are hidden.

NotificationBlock:
- Find or generate a new-type notification about one of your posts; this will begin with "Replied to your post."
- Activate NotificationBlock on the linked post and confirm that the notification is hidden.

Quote Replies:
- Confirm that Quote Replies adds action buttons to "replied to your post", "replied to you in a post", and "mentioned you on a post" notifications. (I do this by enabling responsive design mode and emulating a mobile device, which will show the buttons without hovering them.)
- Confirm that Quote Replies does not add action buttons to "replied to you in a [community name] post" or "mentioned you in a comment in [community name]" notifications.
- Confirm that Quote Replies generates draft posts with the correct contents from all three supported notification types, from both the activity page and activity modal.
- Break the Quote Replies API fetch. Confirm that Quote Replies generates draft posts from "replied to your post" / "replied to you in a post" notifications, with degraded formatting. Confirm that "mentioned you on a post" notifications will instead show an error modal (this code path is unchanged and we didn't write a fallback into it).
- Confirm that the links in the generated draft posts all point to the correct accounts/urls.
